controllers/drplacementcontrol_controller.go

```diff
@@ -1634,32 +1639,67 @@ func (r *DRPlacementControlReconciler) getClusterDecisionFromPlacementRule(plRul
-) *clrapiv1beta1.ClusterDecision {
-	plDecisionName := fmt.Sprintf(PlacementDecisionName, placement.GetName())
-	plDecision := &clrapiv1beta1.PlacementDecision{}
+// getPlacementDecisionFromPlacement returns a PlacementDecision for the passed in Placement if found, and nil otherwise
+// - The PlacementDecision is determined by listing all PlacementDecisions in the Placement namespace filtered on the
+// Placement label as set by OCM
+// - Function also ensures there is only one decision for a Placement, as the needed by the Ramen orchestrators, and
+// if not returns an error
+func (r *DRPlacementControlReconciler) getPlacementDecisionFromPlacement(placement *clrapiv1beta1.Placement,
+) (*clrapiv1beta1.PlacementDecision, error) {
+	matchLabels := map[string]string{
+		clrapiv1beta1.PlacementLabel: placement.GetName(),
+	}
 
-	err := r.Get(context.TODO(), types.NamespacedName{Name: plDecisionName, Namespace: placement.Namespace}, plDecision)
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			r.Log.Info("Failed to retrieve PlacementDecision", "pdName", plDecisionName)
-		}
+	listOptions := []client.ListOption{
+		client.InNamespace(placement.GetNamespace()),
+		client.MatchingLabels(matchLabels),
+	}
 
-		return &clrapiv1beta1.ClusterDecision{}
+	plDecisions := &clrapiv1beta1.PlacementDecisionList{}
+	if err := r.List(context.TODO(), plDecisions, listOptions...); err != nil {
+		return nil, fmt.Errorf("failed to list PlacementDecisions (placement: %s)",
+			placement.GetNamespace()+"/"+placement.GetName())
 	}
 
+	if len(plDecisions.Items) == 0 {
+		return nil, nil
+	}
+
+	if len(plDecisions.Items) > 1 {
+		return nil, fmt.Errorf("multiple PlacementDecisions found for Placement (count: %d, placement: %s)",
+			len(plDecisions.Items), placement.GetNamespace()+"/"+placement.GetName())
+	}
+
+	plDecision := plDecisions.Items[0]
 	r.Log.Info("Found ClusterDecision", "ClsDedicision", plDecision.Status.Decisions)
 
-	var clusterName, reason string
-	if len(plDecision.Status.Decisions) > 0 {
-		clusterName = plDecision.Status.Decisions[0].ClusterName
-		reason = plDecision.Status.Decisions[0].Reason
+	if len(plDecision.Status.Decisions) > 1 {
+		return nil, fmt.Errorf("multiple placements found in PlacementDecision"+
+			" (count: %d, Placement: %s, PlacementDecision: %s)",
+			len(plDecision.Status.Decisions),
+			placement.GetNamespace()+"/"+placement.GetName(),
+			plDecision.GetName()+"/"+plDecision.GetNamespace())
 	}
 
-	return &clrapiv1beta1.ClusterDecision{
-		ClusterName: clusterName,
-		Reason:      reason,
+	return &plDecision, nil
+}
+
+// getClusterDecisionFromPlacement returns the cluster decision for a given Placement if found
+func (r *DRPlacementControlReconciler) getClusterDecisionFromPlacement(placement *clrapiv1beta1.Placement,
+) *clrapiv1beta1.ClusterDecision {
+	plDecision, err := r.getPlacementDecisionFromPlacement(placement)
+	if err != nil {
+		// TODO: err ignored by this caller
```

Comment:
We should create an issue to fix this. The error was disregarded by the caller in the past, and now it continues to be ignored. However, currently, it merely involves a basic `Get` operation. With this PR, we have three distinct instances where an error might be returned. Nonetheless, I think we should address this independently from this PR.

Reply:
Agree, avoided it as returning an error here caused a few more places to change and so left it as is for now. Will add an issue with a ref to code once this is merged.